28.py

```diff
@@ -0,0 +1,26 @@
+# n = len(haystack) - len(needle) + 1
+# m = len(needle)
+# time: O(nm)
+# space: O(1)
+
+class Solution(object):
+    def strStr(self, haystack: str, needle: str) -> int:
+        """A function to return the index of the first occurrence of needle in haystack.
+        Args:
+            haystack (str): A given a string.
+            needle (str): A given a string.
+        Returns:
+            int: The index of the first occurrence of needle in haystack, or -1 if needle is not part of haystack, or 0 if needle is an empty string.
+        """
+        
+        if len(needle) == 0:
```

Comment:
if not needle
https://qiita.com/simonritchie/items/bb06a7521ae6560738a7#%E6%9D%A1%E4%BB%B6%E6%96%87%E3%81%AE%E6%9B%B8%E3%81%8D%E6%96%B9